Sourcery/Parsing/Utils/AnnotationsParser.swift

```diff
@@ -260,7 +260,21 @@ internal struct AnnotationsParser {
                         value = String(value[value.index(after: value.startIndex) ..< value.index(before: value.endIndex)])
                         value = value.trimmingCharacters(in: .whitespaces)
                     }
-                    append(key: name, value: value as NSString, to: &annotations)
+
+                    guard let data = (value as String).data(using: .utf8) else {
+                        append(key: name, value: value as NSString, to: &annotations)
+                        return
+                    }
+
+                    if let jsonArray = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [[String: Any]],
```

Comment:
why not to parse it just once and then do type casting? This would be more performant and compact. I also wonder if conditional typecasting to NSArray and NSDictionary is really needed, aren't they bridged automatically?

```swift
guard let data = (value as String).data(using: .utf8),
   let json = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) else {
  append(key: name, value: value as NSString, to: &annotations)
  return
}
if let array = json as? [[String: Any]] {
  append(key: name, value: json as! NSArray, to: &annotations)
} else if let dict = json as? [String: Any] {
  append(key: name, value: json as! NSDictionary, to: &annotations)
} else {
  append(key: name, value: value as NSString, to: &annotations)
}
```

Reply:
true. much better.